src/Core/Domain/ImageSettings/CommandHandler/DeleteImagesFromTypeHandler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\ImageSettings\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\DeleteImagesFromTypeCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageNotDeletedException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeNotFoundException;
+use PrestaShopBundle\Entity\Repository\ImageTypeRepository;
+
+/**
+ * Handles command that delete images from defined image type
+ */
+#[AsCommandHandler]
+final class DeleteImagesFromTypeHandler implements DeleteImagesFromTypeHandlerInterface
+{
+    public function __construct(
+        private readonly ImageTypeRepository $imageTypeRepository
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DeleteImagesFromTypeCommand $command): void
+    {
+        // Get image type by id
+        $imageType = $this->imageTypeRepository->find($command->getImageTypeId()->getValue());
+
+        if (null === $imageType->getId()) {
+            throw new ImageTypeNotFoundException(sprintf('Unable to find image type with id "%d" for deletion', $command->getImageTypeId()->getValue()));
+        }
+
+        // Delete all images linked to image type
+        $this->deleteImagesFromType('small_default', _PS_IMG_DIR_ . '{c,m,su,p,st}/');
```

Comment:
Ah so there is a dedicated command to delete the images associated to an image type so it's good But that's a feature in itself, the image type is not removed here It's kid of like removing some cache files because they will be regenerated dynamically later right?

Then the question remains, should the images be removed along with the image type when it's deleted. Good news is that if it should the code that handles it already exist